gretl/src/main/java/ch/so/agi/gretl/tasks/Publisher.java

```diff
@@ -42,70 +33,244 @@
+
+    private Object validationConfig=null; // Konfiguration fuer die Validierung (eine ilivalidator-config-Datei) z.B. "validationConfig.ini"
+
+    private Boolean userFormats=false; // Benutzerformat (Geopackage, Shapefile, Dxf) erstellen
+
+    private Endpoint kgdiService=null; // Endpunkt des SIMI-Services
+
+    private Endpoint kgdiTokenService=null; // Endpunkt des Authentifizierung-Services
+
+    private Object grooming=null; // Konfiguration fuer die Ausduennung z.B. "grooming.json"
+
+    private String exportModels=null; // Das Export-Modell, indem die Daten exportiert werden
+
+    private String modeldir=null;     // Dateipfade, die Modell-Dateien (ili-Dateien) enthalten
+
+    private String proxy=null;        // Proxy Server fuer den Zugriff auf Modell Repositories
+
+    private Integer proxyPort=null;    // Proxy Port fuer den Zugriff auf Modell Repositories
+
+    private Date version=null;
     @Input
-    public String dataIdent=null; // Identifikator der Daten z.B. "ch.so.agi.vermessung.edit"
+    public String getDataIdent() {
+        return dataIdent;
+    }
     @Input
-    public Endpoint target=null; // Zielverzeichnis
-    @InputFile
-    @Optional
-    public Object sourcePath=null; // Quelldatei z.B. "/path/file.xtf"
+    public Endpoint getTarget() {
+        return target;
+    }
     @Input
     @Optional
-    public Connector database=null; //  Datenbank mit Quelldaten z.B. ["uri","user","password"]. Alternative zu sourcePath
+    public Object getSourcePath() {
+        return sourcePath;
+    }
     @Input
     @Optional
-    public String dbSchema=null; // Schema in der Datenbank z.B. "av"
+    public Connector getDatabase() {
+        return database;
+    }
     @Input
     @Optional
-    public String dataset=null; //  ili2db-Datasetname der Quelldaten "dataset"
+    public String getDbSchema() {
+        return dbSchema;
+    }
     @Input
     @Optional
-    public String modelsToPublish=null; //  ili2db-Modellname(n) zur Auswahl der Quelldaten
+    public String getDataset() {
+        return dataset;
+    }
+
     @Input
     @Optional
-    public String region=null; // Muster der der Dateinamen oder Datasetnamen, falls die Publikation Regionen-weise erfolgt z.B. "[0-9][0-9][0-9][0-9]"     
+    public String getModelsToPublish() {
+        return modelsToPublish;
+    }
+
     @Input
     @Optional
-    public ListProperty<String> regions=null; // Liste der zu publizierenden Regionen (Dateinamen oder Datasetnamen). Nur falls die Publikation Regionen-weise erfolgen soll     
-    private ListProperty<String> _publishedRegions=getProject().getObjects().listProperty(String.class);
+    public String getRegion() {
+        return region;
+    }
+
+    @Input
     @Optional
+    public ListProperty<String> getRegions() {
+        return regions;
+    }
+
+
+    @Internal
```

Comment:
@Optional muss neu mit z.B. @Input kombiniert werden (https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Optional.html). Ich habe dieses Property mit @Internal markiert, da aus meiner Sicht @Input hier nicht passt. Das Property wird immer auf Basis anderer Properties (Regions/Region) gesetzt. Für den Up-To-Date Check ist es daher nicht nötig, dieses auch noch zu berücksichtigen, da es sich gezwungenermassen ändert, sobald sich Region oder Regions auch ändert.